interp.py

```diff
@@ -0,0 +1,27 @@
+import jax
+import jax.numpy as jnp
+from functools import partial
+from jax.config import config 
+config.update("jax_enable_x64", True)
+
+@jax.jit
+def jit_map_coordinates(vals, coords):
+    return jax.scipy.ndimage.map_coordinates(vals, coords, order=1, mode='nearest')
+
+@partial(jax.jit, static_argnames=['dim'])
+def vals_to_coords(grids, x_vals, dim):
+    """Transform values of the states to corresponding coordinates (array indices) on the grids.
+    """
+    intervals = jnp.asarray([grid[1] - grid[0] for grid in grids])
+    low_bounds = jnp.asarray([grid[0] for grid in grids]) 
+    intervals = intervals.reshape(dim, 1)
+    low_bounds = low_bounds.reshape(dim, 1)
+    return (x_vals - low_bounds) / intervals
```

Comment:
Thanks, @HumphreyYang. We can remove `dim` completely and so no partial is required. Just change lines 17 and 18 to:
```
intervals = intervals.reshape(-1, 1)
low_bounds = low_bounds.reshape(-1, 1)
```

Reply:
@Smit-create Interesting. I'm pretty sure I wrote something like this before and it didn't compile.